desktop-exporter/app/components/waterfall-view/waterfall-view.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useRef, ClipboardEvent } from "react";
+import { FixedSizeList } from "react-window";
+import {
+  Text,
+  Flex,
+  Heading,
+  Spacer,
+  useColorModeValue,
+} from "@chakra-ui/react";
+import { useSize } from "@chakra-ui/react-use-size";
+
+import { SpanData } from "../../types/api-types";
+
+const waterfallItemHeight = 50;
+const headerRowHeight = 30;
+const spanNameColumnWidth = 244;
+const serviceNameColumnWidth = 120;
+
+type WaterfallRowProps = {
+  index: number;
+  style: React.CSSProperties;
+  data: WaterfallViewProps;
+};
+
+function WaterfallRow({ index, style, data }: WaterfallRowProps) {
+  let { spans, selectedSpanID, setSelectedSpanID } = data;
+  let span = spans[index];
+
+  // Set the background colour to make the list striped.
+  let backgroundColour =
+    index % 2 ? "" : useColorModeValue("gray.50", "gray.700");
+  let selectedColour = useColorModeValue("pink.50", "pink.900");
+
+  //Set the style for the selected item
+  if (!!selectedSpanID && selectedSpanID === span.spanID) {
+    backgroundColour = selectedColour;
+  }
+
+  // Add zero-width space after forward slashes, dashes, and dots
+  // to indicate line breaking opportunity
+  let nameLabel = span.name
+    .replaceAll("/", "/\u200B")
+    .replaceAll("-", "-\u200B")
+    .replaceAll(".", ".\u200B");
+
+  let stripZeroWidthSpaces = (e: ClipboardEvent<HTMLParagraphElement>) => {
+    let selection = window.getSelection();
+    if (!selection) {
+      return;
+    }
+    let text = selection.toString().replaceAll("\u200B", "");
+    e.clipboardData?.setData("text/plain", text);
+    e.preventDefault();
+  };
```

Comment:
Since this closure doesn't capture any context, let's extract it as a defined function

```js
function stripZeroWidthSpacesOnCopyCallback(e: ClipboardEvent<HTMLParagraphElement>) {
  let selection = window.getSelection();
  if (!selection) {
    return;
  }
  let text = selection.toString().replaceAll("\u200B", "");
  e.clipboardData?.setData("text/plain", text);
  e.preventDefault();  
}
```